### PR TITLE
Add egress-router-cni init container

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds a new init container for the deployment of the
egress-router-cni plugin to the hosts.

### CODE REVIEW

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 914a2e0998e7cc6723bac3efa3e9ff74a51ca87b

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 03c5af013519f3c120f4e18e78340f0fe164128d

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 03c5af013519f3c120f4e18e78340f0fe164128d

**Review by dougbtv — CHANGES_REQUESTED** [review]
@danielmellado don't forget the image-references updates too here @ https://github.com/openshift/cluster-network-operator/blob/master/manifests/image-references

File: bindata/network/multus/multus.yaml [hunk 1]
```diff
@@ -133,6 +133,24 @@ spec:
           value: "/usr/src/multus-cni/rhel8/bin/"
         - name: DEFAULT_SOURCE_DIRECTORY
           value: "/usr/src/multus-cni/bin/"
+      - name: egress-router-binary-copy
+        image: "quay.io/dmellado/egress-router-test:latest"
```
Comment (dougbtv) [inline bindata/network/multus/multus.yaml:137]: Make sure this winds up using the templated `{{.ImageNameHere}}` when you're ready to take this out of WIP

File: manifests/0000_70_cluster-network-operator_03_deployment.yaml [hunk 2]
```diff
@@ -53,6 +53,8 @@ spec:
           value: "5000"
         - name: OVN_CONTROLLER_INACTIVITY_PROBE
           value: "30000"
+        - name: EGRESS_ROUTER_IMAGE
+          value: "quay.io/dmellado/egress-router-test:latest"
```
Comment (dougbtv) [inline manifests/0000_70_cluster-network-operator_03_deployment.yaml:57]: I realize this is probably why it's a WIP, but, yeah, make sure you update to the actual image when taking it out of WIP.

**Review by dougbtv — CHANGES_REQUESTED** [review]

**Comment by danwinship** [comment]
I had been meaning to add https://issues.redhat.com/browse/SDN-695 to the 4.7 epics, which I have belatedly done now. We _really_ need a cleaner system for installing CNI plugins. I had wanted to say that solving this problem would block the addition of any new initcontainers in CNO in 4.7, though ideally I would have say that _before_ you wrote this PR...

**Comment by danielmellado** [comment]
> I had been meaning to add https://issues.redhat.com/browse/SDN-695 to the 4.7 epics, which I have belatedly done now. We _really_ need a cleaner system for installing CNI plugins. I had wanted to say that solving this problem would block the addition of any new initcontainers in CNO in 4.7, though ideally I would have say that _before_ you wrote this PR...

Given that I'm basically relying on multus' daemonset here, I was hoping that we could bypass that for now.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 531c01001b9d6896943f371e375ffa117eed3125

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now f9b07b48fc016f98434e0cf5206a5fc8d17020fd

**Comment by danielmellado** [comment]
Until we get the image in openshift, we can't really update the image-references and this would fail. Setting this to HOLD until the comet request is approved.

File: manifests/image-references [hunk 3]
```diff
@@ -42,6 +42,11 @@ spec:
     from:
       kind: DockerImage
       name: quay.io/openshift/origin-ovn-kubernetes:latest
+# NOTE: Skipped until the image is added to an actual release
```
Comment (dougbtv) [inline manifests/image-references:45]: I think this is what's causing a CI failure, you probably need this.

But, now it's going to probably complain that it's an image that's external to the release.

Comment (danielmellado) [reply manifests/image-references:45]: It was complaining about not being able to find the image nevertheless before that. From my conversations with @squeed it does seem that that's an alias that searches for the actual url, once the image is on a release, so probably stuck on that dependency.

Comment (danielmellado) [reply manifests/image-references:45]:
``` info: Included 122 images from 35 input operators into the release
error: unable to create a release: operator "cluster-network-operator" contained an invalid image-references file: no input image tag named "egress-router-cni" ``` This was the issue with that block before putting that out.

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now 205b864a58856c604e887809f24eaa9d4b1846f3

**Comment by danielmellado** [comment]
Pull-request updated, HEAD is now b23cce2d81734ad0249f56e056d5d90075044acf

**Comment by tssurya** [comment]
/lgtm

note to self: change adds the new egress-router-container image, similar to existing ones.

File: manifests/0000_70_cluster-network-operator_03_deployment.yaml [hunk 4]
```diff
@@ -55,6 +55,8 @@ spec:
           value: "5000"
         - name: OVN_CONTROLLER_INACTIVITY_PROBE
           value: "30000"
+        - name: EGRESS_ROUTER_CNI_IMAGE
+          value: "quay.io/openshift/origin-egress-router-cni:latest"
```
[the remark this answers is not in the document]
Comment (juanluisvaladas) [reply manifests/0000_70_cluster-network-operator_03_deployment.yaml:59]: Not very fond of the idea of having it pointing to latest, but every other image is doing it so I guess that's not terrible

**Pull request merged**